script/cargo.py

```diff
@@ -0,0 +1,143 @@
+
+    # Validate rustup_bin args
+    if (args.rustup_bin is None or
+        len(args.rustup_bin) is not 1 or
+            len(args.rustup_bin[0]) is 0):
+        raise Exception("rustup_bin argument was not specified correctly")
+
+    # Validate rustup_home args
+    if (args.rustup_home is None or
+        len(args.rustup_home) is not 1 or
+            len(args.rustup_home[0]) is 0):
+        raise Exception("rustup_home argument was not specified correctly")
+
+    # Validate cargo_home args
+    if (args.cargo_home is None or
+        len(args.cargo_home) is not 1 or
+            len(args.cargo_home[0]) is 0):
+        raise Exception("cargo_home argument was not specified correctly")
+
+    # Validate manifest_path args
+    if (args.manifest_path is None or
+        len(args.manifest_path) is not 1 or
+            len(args.manifest_path[0]) is 0):
+        raise Exception("manifest_path argument was not specified correctly")
+
+    # Validate build_path args
+    if (args.build_path is None or
+        len(args.build_path) is not 1 or
+            len(args.build_path[0]) is 0):
+        raise Exception("build_path argument was not specified correctly")
+
+    if "out" not in args.build_path[0]:
+        raise Exception("build_path did not contain 'out'")
+
+    # Validate platform args
+    if (args.platform is None or
+        len(args.platform) is not 1 or
+            len(args.platform[0]) is 0):
+        raise Exception("platform argument was not specified correctly")
+
+    # Validate is_debug args
+    if (args.is_debug is None or
+        len(args.is_debug) is not 1 or
+            len(args.is_debug[0]) is 0):
+        raise Exception("is_debug argument was not specified correctly")
+
+    if (args.is_debug[0] != "false" and args.is_debug[0] != "true"):
+        raise Exception("is_debug argument was not specified correctly")
+
+    # args are valid
+    return args
+
+
+def build(rustup_bin, rustup_home, cargo_home, manifest_path, build_path, platform, is_debug):
+    targets = []
+
+    if platform == "Windows x86":
+        targets = ["i686-pc-windows-msvc"]
+    elif platform == "Windows x64":
+        targets = ["x86_64-pc-windows-msvc"]
+    elif platform == "macOS x64":
+        targets = ["x86_64-apple-darwin"]
+    elif platform == "Linux x64":
+        targets = ["x86_64-unknown-linux-gnu"]
+    elif platform == "Android arm":
+        targets = ["arm-linux-androideabi"]
+    elif platform == "Android arm64":
+        targets = ["aarch64-linux-android"]
+    elif platform == "Android x86":
+        targets = ["i686-linux-android"]
+    elif platform == "Android x64":
+        targets = ["x86_64-linux-android"]
+    elif platform == "iOS":
+        targets = ["aarch64-apple-ios", "x86_64-apple-ios"]
+    else:
+        raise ValueError('Cannot build due to unknown platform')
+
+    # Set environment variables for rustup
+    env = os.environ.copy()
+    env['PATH'] += os.pathsep + rustup_bin
```

Comment:
this adds rustup_bin to the end of the path so it won't work if there's another rust install, but I think it would be better to pass the full path which I guess would actually be `cargo_bin` instead of `rustup_bin` and call that directly in line 127 instead of changing the path

Reply:
or maybe do both and prepend `rustup_bin_path` and then call rustup_bin_path + os.pathsep + "cargo" below

Reply:
we could pass the full path but it will not help with rustc etc. so I have fixed the bug where rustup_bin was passed to the end of the path and now prepend which is working for local and brave rust installs